compute/cker/src/train/BinaryArithmetic.test.cc

```diff
@@ -58,3 +58,43 @@ TEST(CKer_Operation, AddGrad)
       EXPECT_FLOAT_EQ(rhs_backward[i], expected_rhs_backward[i]);
   }
 }
+
+TEST(CKer_Operation, SubGrad)
+{
+  {
+    // Shape: {2, 3}
+    std::vector<float> incoming_backward = {-2, 3, -4, 5, -6, 7};
+    std::vector<float> expected_lhs_backward = incoming_backward;
+    std::vector<float> expected_rhs_backward = incoming_backward;
+    std::vector<float> lhs_backward(6);
+    std::vector<float> rhs_backward(6);
+
+    nnfw::cker::train::BinaryArithmeticGrad(nnfw::cker::Shape{2, 3}, incoming_backward.data(),
+                                            nnfw::cker::Shape{2, 3}, lhs_backward.data(),
```

Comment:
@Aeren1564 Did you test with different shape of `lhs` and `rhs`. For example, it would be good to have a test case with `lhs.shape = (1,3)`. Also, what happens when it is not correct shape (that is `lhs.shape = (3,3)`) ? Do we assume its shape is always correct because forward pass will be done before backward pass?

Reply:
Is it possible to have `lhs` and `rhs` with different shapes?

I assumed that we'll always have valid shape because [tf.keras.layers.Subtract](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Subtract) specification stated that

```It takes as input a list of tensors of size 2, both of the same shape, and returns a single tensor, (inputs[0] - inputs[1]), also of the same shape.```

so I thought that we can't have input with different shapes.

Reply:
> Is it possible to have lhs and rhs with different shapes?

Yes, it is possible. But, considering the broadcasting of Eigen(https://eigen.tuxfamily.org/dox/group__TutorialReductionsVisitorsBroadcasting.html), The current kernel does not support broadcasting yet.

Reply:
By the way, I guess the current kernel supports only 2D. If not, could you add tests with rank higher than 3?

Reply:
1. The following check has been added.
```cpp
  if (lhs_grad_shape != rhs_grad_shape)
    throw std::runtime_error{"Shape of LHS and RHS must match"};
```

2. Unittest for distinct shape has been added.

> By the way, I guess the current kernel supports only 2D. If not, could you add tests with rank higher than 2?

3. Unittest for dimension $2x2x2$ has been added.

Reply:
Could you also add checking shape of incoming?

Reply:
The check has been modified to

```cpp
  if (lhs_grad_shape != rhs_grad_shape || lhs_grad_shape != incoming_shape ||
      rhs_grad_shape != incoming_shape)
    throw std::runtime_error{"Shape of LHS, RHS, and incoming must match"};
```